collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdapter.java

```diff
@@ -78,15 +89,27 @@
     private boolean playerIsReady;
     private boolean isVideoAttemptedPlay = false;
     private DRMInformation drmInformation = null;
+    private BitmovinSegmentTrackingAdapter segmentTrackingAdapter;
+    private BitmovinErrorDetailsAdapter errorDetailsAdapter;
+    private Context context;
+    private BitmovinAnalytics analytics;
 
-    public BitmovinSdkAdapter(BitmovinPlayer bitmovinPlayer, BitmovinAnalyticsConfig config, EventDataFactory factory, PlayerStateMachine stateMachine) {
+    public BitmovinSdkAdapter(BitmovinAnalytics analytics, BitmovinPlayer bitmovinPlayer, BitmovinAnalyticsConfig config, EventDataFactory factory, PlayerStateMachine stateMachine, Context context) {
+        this.analytics = analytics;
         this.config = config;
         this.stateMachine = stateMachine;
         this.bitmovinPlayer = bitmovinPlayer;
         this.factory = factory;
+        this.context = context;
     }
 
     public void init() {
+        this.segmentTrackingAdapter = new BitmovinSegmentTrackingAdapter(this.bitmovinPlayer);
```

Comment:
i'd move all of this out of here into a factory. it will get more and more over time so starting to seperate object creation from usage would be a good idea anyways.
Might make sense to look at DI Containers that manage object lifecycle like Guice or this other thing that the player ppl are starting to use on Android

Reply:
Do we really want to pull in external dependencies?
How would that look like with a factory? Just something simple like a `BitmovinFeaturesFactory.create(player): List<Features>` where we create all that stuff? So only moving it out from here?